src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax;
+
+public class Rectangle extends Figure {
+    private final double firstLeg;
+    private final double secondLeg;
+    private final Color color;
+
+    public Rectangle(Color color, double firstLeg, double secondLeg) {
+        this.color = color;
+        this.firstLeg = firstLeg;
+        this.secondLeg = secondLeg;
+    }
+
+    @Override
+    public double calculateArea() {
+        return firstLeg * secondLeg;
+    }
+
+    @Override
+    public void draw() {
+        System.out.println("Figure: rectangle, area: "
+                + calculateArea() + "sq. units, firstLeg: "
+                + firstLeg + " units, " + "secondLeg: "
+                + secondLeg + " units, "
+                + "color: " + color.name());
```

Comment:
When concatenating strings in the 'draw()' method, it's a good practice to use a StringBuilder to improve performance, especially if the concatenation is done in a loop or in methods that are called frequently.